### ISSUE TITLE
Testing XEP-0368 needs improvement

### ISSUE THREAD

**Issue by herrwusel** [iNPUTmice/caas#19]
The compliance test says that my server supports XEP-0368 but in reality it doesn't. I don't know how you check - I have the legacy ports option on my prosody server defined - but it is not sufficient enough. I can't connect to my server from guest and work networks.

**Comment by licaon-kter** [comment]
Detail your setup... ports route where? Using SSLH? etc

**Comment by herrwusel** [comment]
I'm using prosody 0.10 and sslh isn't used yet. I have SRV records for 433, 5222, 5223 and 5269. Anything else you need?

**Comment by iNPUTmice** [comment]
I don't understand what it use you want the test to do differently. It
checks if the records exists and if there is a server responding on the
announced ports. It obviously can't VPN into your wifi and check it from
there

**Comment by licaon-kter** [comment]
In Conversations (on a good network, not guest/work firewalled) enable Advanced server thing, put domain name and port 443, does it work?

**Comment by herrwusel** [comment]
@iNPUTmice I dont know. Just wanted to inform you that is not a good test since it isn't testing the functionality.

@licaon-kter Did that already and connecting on a unrestricted working to xmpp on port 443 doesn't work.

**Comment by licaon-kter** [comment]
Some stanza entries from logcat could help, maybe it sends some stuff as expected and CaaS registers as available, but it doesn't login because of some other thing hence Conversations fails.

**Comment by herrwusel** [comment]
[Log](https://privatebin.net/?c5bb52966123ce4f#/q1eArFyAlP2H2NEj8SzoAlSf09uT3yXAd5ndSC2Vb4=)
In the beginning of the log port was set to 443, set it back to 5222 and after to 443 again.

**Comment by licaon-kter** [comment]
You need to specify the timestamps where 443 or 5222 was used

Also, there were no DNS messages?

**Comment by herrwusel** [comment]
Sorry, my bad. I pasted everything from the log. New [log](https://privatebin.net/?3928efbbbbd67332#TkKMPTAuwEzaPDeDJYqzNYFlGcZIVI6KpmbrdP7y1/E=):
10-09 12:25:41.060 - port 433
10-09 12:26:43.485 - port 5222

**Comment by herrwusel** [comment]
I set up sslh and still think there is a  problem with the test. What I did:
- set up sslh so the server is reachable through port 443 (conversations is able to connect in a home network/ when setting port manually to 443 connecting is possible)
- rerun test on https://compliance.conversations.im and XEP 0368 is available
- disable sslh (connecting on port 443 fails)
- rerun test on https://compliance.conversations.im and XEP 0368 is wrongfully availabe.

As I tried to communicate before: I don't think that is a behaviour you want to have for an overview page where ppl chosing services according to their features.

**Comment by licaon-kter** [comment]
Your SRVs are setup as here: https://wiki.xmpp.org/web/Tech_pages/XEP-0368 right?

**Comment by herrwusel** [comment]
Yes, is the test only checking for these? If so a simple try to connect should be better imo.
